src/Workspaces/Remote/Core/Services/SolutionService.cs

```diff
@@ -22,15 +24,27 @@ internal class SolutionService : ISolutionController
         private volatile static Tuple<Checksum, Solution> s_primarySolution;
         private volatile static Tuple<Checksum, Solution> s_lastSolution;
 
-        public SolutionService(AssetService assetService, RemoteWorkspace remoteWorkspace)
+        public SolutionService(AssetService assetService)
         {
             _assetService = assetService;
-            PrimaryWorkspace = remoteWorkspace;
         }
 
-        public RemoteWorkspace PrimaryWorkspace
+        public static RemoteWorkspace PrimaryWorkspace
         {
-            get;
+            get
+            {
+                var exportProvider = (IMefHostExportProvider)RoslynServices.HostServices;
+                var primaryWorkspace = exportProvider.GetExports<PrimaryWorkspace>().Single().Value;
```

Comment:
this and code in RoslynServices are dup except HostServices. it would be less confusing who sets PrimaryWorkspace if it is extracted out to like "RemoteWorkspace.GetOrCreate(HostServices)" and reused in both places.

and explain that any one who calls this method first will set PrimaryWorkspace for the HostServices.

Reply:
The caller should not be attempting to use the workspace prior to an explicit initialization sequence somewhere. This PR is just a workaround until the underlying design flaw can be fixed.